packages/openid4vc/src/openid4vc-verifier/OpenId4VcSiopVerifierService.ts

```diff
@@ -93,9 +93,60 @@ export class OpenId4VcSiopVerifierService {
     // Correlation id will be the id of the verification session record
     const correlationId = utils.uuid()
 
+    const requestSigner = options.requestSigner
+
+    let clientId: string
+    let clientIdScheme: ClientIdScheme
+
+    if (requestSigner.method === 'x5c') {
+      const issuer = requestSigner.issuer
+      const leafCertificate = X509Service.getLeafCertificate(agentContext, {
+        certificateChain: requestSigner.x5c,
+      })
+
+      if (issuer.startsWith('dns:')) {
+        // If the iss value contains a DNS name encoded as a URI using the DNS URI scheme [RFC4501],
+        // the DNS name MUST match a dNSName Subject Alternative Name (SAN) [RFC5280] entry of the leaf certificate.
+        const extractDnsNameFromDnsUriScheme = (iss: string): string => {
+          const afterDns = iss.substring('dns:'.length)
+          const withoutQuery = afterDns.split('?')[0]
+          return withoutQuery.split('/').pop() || ''
+        }
+
+        const dnsName = extractDnsNameFromDnsUriScheme(issuer)
+
+        // TODO: HAIP: x.509 certificates: the SD-JWT VC contains the issuer's certificate along with a trust chain in the x5c JOSE header.
+        // In this case, the iss value MUST be an URL with a FQDN matching a dNSName Subject Alternative Name (SAN) [RFC5280] entry in the leaf certificate.
```

Comment:
Is this still todo?

Reply:
yes. with haip it cannot be a san uri name